spynnaker/pyNN/models/neural_projections/connectors/fixed_number_post_connector.py

```diff
@@ -1,104 +1,194 @@
 from .abstract_connector import AbstractConnector
+from spynnaker.pyNN.exceptions import SpynnakerException
 import numpy
 import logging
 
 logger = logging.getLogger(__file__)
 
 
 class FixedNumberPostConnector(AbstractConnector):
+    """ Connects a fixed number of post-synaptic neurons selected at random,
+        to all pre-synaptic neurons
+    """
 
     def __init__(
             self, n, allow_self_connections=True, safe=True, verbose=False):
+        """
+        :param `int` n:
+            number of random post-synaptic neurons connected to output
+        :param `bool` allow_self_connections:
+            if the connector is used to connect a
+            Population to itself, this flag determines whether a neuron is
+            allowed to connect to itself, or only to other neurons in the
+            Population.
+        """
         AbstractConnector.__init__(self, safe, verbose)
-        self._post_n = n
+        self._n_post = n
         self._allow_self_connections = allow_self_connections
-        self._post_neurons = None
+        self._verbose = verbose
+        self._post_neurons_set = False
 
     def get_delay_maximum(self):
         return self._get_delay_maximum(
-            self._delays, self._n_pre_neurons * self._post_n)
+            self._delays, self._n_pre_neurons * self._n_post)
 
     def get_delay_variance(
             self, pre_slices, pre_slice_index, post_slices,
             post_slice_index, pre_vertex_slice, post_vertex_slice):
-        if not self._is_connected(post_vertex_slice):
+        if not self._is_connected(post_vertex_slice, 0):
```

Comment:
Not sure that passing 0 here is enough... might other post neurons be connected?

Reply:
Oops - marked all these this way to start off with and forgot to come back to sort them out for all possible post-neuron connections... will sort this out.